doc/user/content/overview/arangements.md

```diff
@@ -0,0 +1,149 @@
+![Diagram of arrangements for count](/images/arrangements-count.png "Diagram of arrangements for COUNT")
+
+#### Three-way join
+
+Using an example from the [TPC-H](http://www.tpc.org/tpch/) data warehousing benchmark, let's say that you want to determine which ten orders that haven't shipped as of a certain date have the greatest potential revenue. The query itself is:
+
+```sql
+SELECT
+    l_orderkey,
+    o_orderdate,
+    o_shippriority,
+    sum(l_extendedprice * (1 - l_discount)) AS revenue
+FROM
+    customer,
+    orders,
+    lineitem
+WHERE
+    c_mktsegment = 'BUILDING'
+    AND c_custkey = o_custkey
+    AND l_orderkey = o_orderkey
+    AND o_orderdate < DATE '1995-03-15'
+    AND l_shipdate > DATE '1995-03-15'
+GROUP BY
+    l_orderkey,
+    o_orderdate,
+    o_shippriority
+ORDER BY
+    revenue desc,
+    o_orderdate
+LIMIT 10;
+```
+
+The query is a three-way join between `customer`, `orders`, and `lineitem`, followed by a reduction. For the purposes of this explanation, we're going to focus on the join and skip the reduction.
+
+When joining three columns, most dataflows will build an index on the combination of Column 1 + Column 2 + Column 3, but also on every other possible combination of columns: Column 1 + Column 2, Column 1 + Column 3, and Column 2 + Column 3.  Materialize does something different.
+
+Materialize builds indexes on Column 1 + 2 and (Column 1 + Column 2) + Column 3. It can do this because it separates out the work of the join calculation from the work of indexing information or creating arrangements. Each of the original columns and both of the necessary joins create arrangements that can be reused for other queries.
+
+![Diagram of arrangements for a three-way join](/images/arrangements-3-way-join.png "Diagram of arrangements for a three-way join")
+
+You can find a more detailed analysis of the dataflow (and the performance benefits of using Materialize) in our blog post on [Joins in Materialize](https://materialize.com/joins-in-materialize).
+
+## Analyzing arrangements
+
+Materialize provides various tools that allow you to analyze arrangements, although they are post hoc tools best used for debugging, rather than planning tools to be used before creating indexes or views. See [Diagnosing Using SQL](/ops/diagnosing-using-sql/) and [Monitoring](/ops/monitoring/) for more details.
+
+## Reducing memory usage
+
+### Creating indexes manually
+
+You can't create arrangements directly, but in some cases you can manually create indexes that will help Materialize create more sharable arrangements -- and therefore reduce memory usage.
+
+For maximal control, you can create an unmaterialized view, then create an index on the view for the keys you know you'll want to search on. This can create a significant memory reduction for cases where Materialize is unable to detect the primary key, for example, if you have a customerID but never actually use it and always search on the join of customer first and last names.
```

Comment:
If Materialize cannot detect a key for a source/view/table, then in order to ensure good data distribution, the default key is the full set of columns. Creating an unmaterialized view and then specifying a custom index makes each key smaller.

Reply:
Replaced the second paragraph.